src/main/java/uk/co/tfn/HelperMethods.java

```diff
@@ -433,33 +438,97 @@ public void completeUserDetailsPage() {
-                        .waitForElement(String.format("passenger-type-%s", String.valueOf(randomUserType)));
+        if(randomSelector == 1){
+            // Click Any and continue
+            WebElement element = this.waitForElement("passenger-type-anyone");
+            if (this.browser.equals("ie")) {
+                JavascriptExecutor executor = (JavascriptExecutor) this.driver;
+                executor.executeScript("arguments[0].click();", element);
+            } else {
+                element.click();
+            }
+            this.continueButtonClick();
+            this.waitForPageToLoad();
+        } else if (randomSelector == 2){
+            // Click Group, complete following pages, and continue
+            WebElement element = this.waitForElement("passenger-type-group");
+            if (this.browser.equals("ie")) {
+                JavascriptExecutor executor = (JavascriptExecutor) this.driver;
+                executor.executeScript("arguments[0].click();", element);
+            } else {
+                element.click();
+            }
+            this.continueButtonClick();
+            this.waitForPageToLoad();
+            this.completeGroupPassengerDetailsPages();
+        } else {
+            // Click a non-Any non-Group, complete the next page, and continue
+            List<WebElement> otherPassengerTypes = new ArrayList<WebElement>();
 
-                if (this.browser.equals("ie")) {
-                    JavascriptExecutor executor = (JavascriptExecutor) this.driver;
-                    executor.executeScript("arguments[0].click();", element);
-                } else {
-                    element.click();
-                }
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-adult")));
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-child")));
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-infant")));
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-senior")));
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-student")));
+            otherPassengerTypes.add(this.driver.findElement(By.id("passenger-type-youngPerson")));
 
-                this.continueButtonClick();
-                this.waitForPageToLoad();
-                this.completeUserDetailsPage();
-                break;
+            WebElement chosenPassenger = otherPassengerTypes.get(randomNumberBetweenOneAnd(6)-1);
+
+            if (this.browser.equals("ie")) {
+                JavascriptExecutor executor = (JavascriptExecutor) this.driver;
+                executor.executeScript("arguments[0].click();", chosenPassenger);
+            } else {
+                chosenPassenger.click();
+            }
+            this.continueButtonClick();
+            this.waitForPageToLoad();
+            this.completeUserDetailsPage(false, 0);
+        }
+    }
+
```

Comment:
Should the below methods live in `StepMethods.java `instead?

Reply:
Think its ok where it is, its in an else block so we only want it to run in a certain scenario.